ext/ClimaCommsCUDAExt.jl

```diff
@@ -50,4 +50,63 @@ ClimaComms.elapsed(f::F, ::CUDADevice, args...; kwargs...) where {F} =
 ClimaComms.assert(::CUDADevice, cond::C, text::T) where {C, T} =
     isnothing(text) ? (CUDA.@cuassert cond()) : (CUDA.@cuassert cond() text())
 
+cuda_thread_index_1D() =
+    (CUDA.blockIdx().x - 1) * CUDA.blockDim().x + CUDA.threadIdx().x
+
+function cuda_device_limits_1D()
+    cuda_device = CUDA.device()
+    max_blocks =
+        CUDA.attribute(cuda_device, CUDA.DEVICE_ATTRIBUTE_MAX_GRID_DIM_X)
+    max_threads_in_block =
+        CUDA.attribute(cuda_device, CUDA.DEVICE_ATTRIBUTE_MAX_THREADS_PER_BLOCK)
+    return (; max_blocks, max_threads_in_block)
+end
+
+function threaded(f::F, ::CUDADevice, ::Val, itr) where {F}
+    length(itr) > 0 || return nothing
+
+    Base.require_one_based_indexing(itr)
+    function call_f_from_thread()
+        item_index = cuda_thread_index_1D()
+        item_index <= length(itr) && @inbounds f(itr[item_index])
+        return nothing
+    end
+    kernel = CUDA.@cuda launch=false call_f_from_thread()
+    optimal_threads_in_block = CUDA.launch_configuration(kernel.fun).threads
```

Comment:
This is just the maximum number of threads. This need not be optimal number of threads.